snuba_sdk/legacy.py

```diff
@@ -233,7 +235,7 @@ def json_to_snql(body: Mapping[str, Any], entity: str) -> Query:
 
     assert isinstance(query.match, Entity)
     time_column = get_required_time_column(query.match.name)
-    if time_column:
+    if time_column and not skip_time:
```

Comment:
I think we should reject a query that tries to set time range conditions for subscriptions. Not silently drop those conditions.

Reply:
This isn't dropping anything. It is preventing the SDK from automatically adding them.

Reply:
Can't the client decide not to add `from_date` or `to_date` to the query body?
The loop below only adds the conditions if those fields are in the query body in the first place.
If no `from_date` and `to_date` are present it would not add anything, so `if skip_time`, we would ignore those fields.

Shouldn't we reject the query if the user provided `from_date` and `to_date` in the legacy query ?

Reply:
Oh I see what you're saying now. You're right this change isn't needed. If someone tries to add a time condition to a subscription, then the validation in Snuba will catch it and error. So the SDK can work as normal.